src/realm/list.hpp

```diff
@@ -154,26 +154,100 @@ class Lst final : public CollectionBaseImpl<LstBase> {
         return *m_tree;
     }
 
+    UpdateStatus update_if_needed() const final
+    {
+        auto status = Base::update_if_needed();
+        switch (status) {
+            case UpdateStatus::Detached: {
+                m_tree.reset();
+                return UpdateStatus::Detached;
+            }
+            case UpdateStatus::NoChange:
+                if (m_tree && m_tree->is_attached()) {
+                    return UpdateStatus::NoChange;
+                }
+                // The tree has not been initialized yet for this accessor, so
+                // perform lazy initialization by treating it as an update.
+                [[fallthrough]];
+            case UpdateStatus::Updated: {
+                bool attached = init_from_parent(false);
+                return attached ? UpdateStatus::Updated : UpdateStatus::Detached;
+            }
+        }
+        REALM_UNREACHABLE();
+    }
+
+    UpdateStatus ensure_writable(bool allow_create) final
```

Comment:
I would rather introduce a function called `ensure_created` and only make a call to that in places where subsequent actions requires an existing bplustree - like `insert`. In other places like `remove` you implicitly call `size()` that will ensure that the object is updated. It is no longer necessary to make the owning m_obj writable. That should be handled when the ref is set from possible `copy_on_write`.